Altis_Life.Altis/core/functions/TON/fn_sendMessage.sqf

```diff
@@ -0,0 +1,69 @@
+/*
+    File: fn_sendMessage.sqf
+    Author: blackfisch
+
+    Description: send message to target
+*/
+params [
+    ["_ehParams", [], [[]]],
+    ["_target", "player", [""]]
+];
+_ehParams params [
+    ["_control", controlNull, [controlNull]]
+];
+
+_control ctrlEnable false;
+
+private _to = call compile format ["%1",(lbData[3004,(lbCurSel 3004)])];
+if ((_target in ["adminToPlayer", "player"]) && {isNil "_to" || {isNull _to}}) exitWith {
+    _control ctrlEnable true;
+};
+
+private _msg = ctrlText 3003;
+if (_msg isEqualTo "") exitWith {
+    hint localize "STR_CELLMSG_EnterMSG";
+    _control ctrlEnable true;
+};
+
+private _maxLength = getNumber(missionConfigFile >> "Life_Settings" >> "message_maxlength");
+if (_maxLength > -1 && {(count _msg) > _maxLength}) exitWith {
+    hint localize "STR_CELLMSG_LIMITEXCEEDED";
+    _control ctrlEnable true;
+};
+
+private _targetFunction = 0;
+private _sender = player getVariable ["realName", profileName];
+private _toName = (if (isNil "_to" || {isNull _to}) then {"ERROR"} else {_to getVariable ["realname", name _to]});
+private _confirmMessage = "STR_CELLMSG_ToPerson";
+
+switch (_target) do {
+    case "cop": {
+        _to = west;
+        _targetFunction = 1;
+        _confirmMessage = "STR_CELLMSG_ToPolice";
+    };
+    case "med": {
+        _to = independent;
+        _targetFunction = 5;
+        _confirmMessage = "STR_CELLMSG_ToEMS";
+    };
+    case "admin": {
+        _to = -2;
+        _targetFunction = 2;
+        _confirmMessage = "STR_CELLMSG_ToAdmin";
+    };
+    case "adminToPlayer": {
+        _targetFunction = 3;
+        _confirmMessage = "STR_CELLMSG_AdminToPerson";
+    };
+    case "adminAll": {
+        _to = -2;
+        _targetFunction = 4;
+        _confirmMessage = "STR_CELLMSG_AdminToAll";
+    };
+};
```

Comment:
Now's a good time to change these magic numbers to something in English?

Either with a lookup table or by changing clientMessage. Seems sensible to use a macro for the RE target too

Reply:
I don't like RE macros :( do I have to? :D

Reply:
The the magic numbers, yes, needs to be readable

Reply:
will replace the targetfunction swtuff with strings for ease

is `-2` and such as RE target ok for you or do you want me to change that to `CLIENT` macro?

Reply:
Just create your own macro at the top, we are phasing those out I believe generally.

It's fine writing -2 in the RE expression, but you're assigning it to a variable in another scope, it's confusing for people to read

```sqf
_re_target = ALL;
```

etc.